src/components/Dashboard/ThisWeek/NewsletterMenu/MenuItems/LogNewsLetter/LogNewsletter.tsx

```diff
@@ -31,28 +56,281 @@ const CloseButton = styled(IconButton)(({ theme }) => ({
   },
 }));
 
-const LogNewsletter = ({ handleClose }: Props): ReactElement<Props> => {
+const LogFormControl = styled(FormControl)(() => ({
+  width: '100%',
+}));
+
+const LogFormLabel = styled(FormLabel)(({ theme }) => ({
+  margin: theme.spacing(1, 0),
+  fontWeight: 'bold',
+  color: theme.palette.primary.dark,
+  '& span': {
+    color: theme.palette.error.main,
+  },
+}));
+
+const LogFormControlLabel = styled(FormControlLabel)(({ theme }) => ({
+  '& span.MuiFormControlLabel-label': {
+    fontWeight: 'bold',
+    color: theme.palette.primary.dark,
+  },
+}));
+
+const LogTextField = styled(TextField)(({ theme }) => ({
+  '& div': {
+    padding: theme.spacing(1),
+  },
+}));
+
+const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
+  margin: theme.spacing(0, 1, 0, 0),
+}));
+
+const taskSchema: yup.SchemaOf<
+  Omit<TaskCreateInput, 'result' | 'nextAction'>
+> = yup.object({
+  id: yup.string().nullable(),
+  activityType: yup.mixed<ActivityTypeEnum>(),
+  subject: yup.string().required(),
+  startAt: yup.string().nullable(),
+  completedAt: yup.string().nullable(),
+  tagList: yup.array().of(yup.string()).default([]),
+  contactIds: yup.array().of(yup.string()).default([]),
+  userId: yup.string().nullable(),
+  notificationTimeBefore: yup.number().nullable(),
+  notificationType: yup.mixed<NotificationTypeEnum>(),
+  notificationTimeUnit: yup.mixed<NotificationTimeUnitEnum>(),
+});
+
+const LogNewsletter = ({
+  accountListId,
+  handleClose,
+}: Props): ReactElement<Props> => {
   const { t } = useTranslation();
 
+  const [commentBody, changeCommentBody] = useState('');
+
+  const [createTask, { loading: creating }] = useCreateTaskMutation();
+  const [createTaskComment] = useCreateTaskCommentMutation();
+
+  const initialTask: TaskCreateInput = {
+    activityType: ActivityTypeEnum.NewsletterPhysical,
+    completedAt: null,
+    contactIds: [],
+    id: null,
+    nextAction: null,
+    notificationTimeBefore: null,
+    notificationTimeUnit: null,
+    notificationType: null,
+    result: null,
+    startAt: DateTime.local().plus({ hours: 1 }).startOf('hour').toISO(),
```

Comment:
What's the logic here? We're just rounding to the next hour for creating the task?

Reply:
Yeah, that seems to be how it's working currently. https://github.com/CruGlobal/mpdx-react/blob/88be98ec93fd87fd56da6d67f7640e3f3beb4e0b/src/components/Task/Drawer/Form/Form.tsx#L128 I had at it ```minus``` which wasn't the best as it would automatically create a late task.